pkg/shares/non_interactive_defaults_test.go

```diff
@@ -0,0 +1,260 @@
+package shares
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/tendermint/tendermint/pkg/consts"
+)
+
+func TestMsgSharesUsedNIDefaults(t *testing.T) {
+	type test struct {
+		cursor, squareSize, expected int
+		msgLens                      []int
+		indexes                      []uint32
+	}
+	tests := []test{
+		{2, 4, 1, []int{1}, []uint32{2}},
+		{2, 2, 1, []int{1}, []uint32{2}},
+		{3, 4, 8, []int{3, 3}, []uint32{4, 8}},
+		{0, 8, 8, []int{8}, nil}, // todo: actually fill out these values instead of nil
+		{0, 8, 7, []int{7}, nil},
+		{0, 8, 7, []int{3, 3}, nil},
+		{1, 8, 8, []int{3, 3}, nil},
+		{1, 8, 32, []int{1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1}, nil},
+		{3, 8, 16, []int{5, 7}, nil},
+		{0, 8, 29, []int{5, 5, 5, 5}, nil},
+		{0, 8, 10, []int{10}, nil},
+		{0, 8, 26, []int{10, 10}, nil},
+		{1, 8, 33, []int{10, 10}, nil},
+		{2, 8, 32, []int{10, 10}, nil},
+		{0, 8, 55, []int{21, 31}, nil},
+		{0, 8, 128, []int{64, 64}, nil},
+		{0, consts.MaxSquareSize, 1000, []int{1000}, nil},
+		{0, consts.MaxSquareSize, consts.MaxSquareSize + 1, []int{consts.MaxSquareSize + 1}, nil},
+		{1, consts.MaxSquareSize, (consts.MaxSquareSize * 4) - 1, []int{consts.MaxSquareSize, consts.MaxSquareSize, consts.MaxSquareSize}, nil},
+		{1024, consts.MaxSquareSize, 32, []int{32}, nil},
```

Comment:
[optional] I filled out these values while reviewing in case they are helpful

```suggestion
		{0, 8, 8, []int{8}, []uint32{0}},
		{0, 8, 7, []int{7}, []uint32{0}},
		{0, 8, 7, []int{3, 3}, []uint32{0, 4}},
		{1, 8, 8, []int{3, 3}, []uint32{2, 6}}, //
		{1, 8, 32, []int{1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1}, []uint32{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32}},
		{3, 8, 16, []int{5, 7}, []uint32{4, 12}},
		{0, 8, 29, []int{5, 5, 5, 5}, []uint32{0, 8, 16, 24}},
		{0, 8, 10, []int{10}, []uint32{0}},
		{0, 8, 26, []int{10, 10}, []uint32{0, 16}},
		{1, 8, 33, []int{10, 10}, []uint32{8, 24}},
		{2, 8, 32, []int{10, 10}, []uint32{8, 24}},
		{0, 8, 55, []int{21, 31}, []uint32{0, 24}},
		{0, 8, 128, []int{64, 64}, []uint32{0, 64}},
		{0, consts.MaxSquareSize, 1000, []int{1000}, []uint32{0}},
		{0, consts.MaxSquareSize, consts.MaxSquareSize + 1, []int{consts.MaxSquareSize + 1}, []uint32{0}},
		{1, consts.MaxSquareSize, (consts.MaxSquareSize * 4) - 1, []int{consts.MaxSquareSize, consts.MaxSquareSize, consts.MaxSquareSize}, []uint32{128, 256, 384}},
		{1024, consts.MaxSquareSize, 32, []int{32}, []uint32{1024}},
```

Reply:
[no change needed] visual for `{3, 8, 16, []int{5, 7}, []uint32{4, 12}},`

```ascii
| | | | |5|5|5|5|
|5| | | |7|7|7|7|
|7|7|7| | | | | |
| | | | | | | | |
| | | | | | | | |
| | | | | | | | |
| | | | | | | | |
| | | | | | | | |
```

Reply:
wow, thanks for this!!